yarpctest/recorder/recorder_test.go

```diff
@@ -160,11 +161,19 @@ func (t *testingTMock) Fatal(args ...interface{}) {
 }
 
 func withDisconnectedClient(t *testing.T, recorder *Recorder, f func(raw.Client)) {
+	httpAgent := http.NewAgent()
+	// TODO lifecycle
+
 	clientDisp := yarpc.NewDispatcher(yarpc.Config{
 		Name: "client",
 		Outbounds: yarpc.Outbounds{
 			"server": {
-				Unary: http.NewOutbound("http://localhost:65535"),
+				Unary: http.NewChooserOutbound(
+					single.New(
```

Comment:
i'm guessing that single peer outbounds will be uncommon in practice, and that we’re feeling the pain because we use them extensively in tests. i think we should wait for a customer to complain.